cmd/status/cmdstatus.go

```diff
@@ -66,49 +95,167 @@ type Runner struct {
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Based on the inventory template manifest we look up the inventory
 	// from the live state using the inventory client.
 	identifiers, err := invClient.GetClusterObjs(inv)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// Exit here if the inventory is empty.
-	if len(identifiers) == 0 {
-		_, _ = fmt.Fprint(cmd.OutOrStdout(), "no resources found in the inventory\n")
-		return nil
+	printData := printer.PrintData{
+		Identifiers: object.ObjMetadataSet{},
+		InvNameMap:  make(map[object.ObjMetadata]string),
+		StatusSet:   r.statusSet,
+	}
+
+	for _, obj := range identifiers {
+		// check if the object is under one of the targeted namespaces
+		if _, ok := r.namespaceSet[obj.Namespace]; ok || len(r.namespaceSet) == 0 {
+			// add to the map for future reference
+			printData.InvNameMap[obj] = inv.Name()
+			// append to identifiers
+			printData.Identifiers = append(printData.Identifiers, obj)
+		}
+	}
+	return &printData, nil
+}
+
+// Retrieve a list of inventory object from the cluster
+func (r *Runner) listInvFromCluster() (*printer.PrintData, error) {
+	invClient, err := r.invFactory.NewClient(r.factory)
+	if err != nil {
+		return nil, err
 	}
 
 	// initialize maps in printData
 	printData := printer.PrintData{
-		InvNameMap: make(map[object.ObjMetadata]string),
-		StatusSet:  make(map[string]bool),
+		Identifiers: object.ObjMetadataSet{},
+		InvNameMap:  make(map[object.ObjMetadata]string),
+		StatusSet:   r.statusSet,
 	}
-	for _, obj := range identifiers {
-		// add to the map for future reference
-		printData.InvNameMap[obj] = inv.Name()
-		// append to identifiers
-		printData.Identifiers = append(printData.Identifiers, obj)
+
+	identifiersMap, err := invClient.ListClusterInventoryObjs(r.ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for invName, identifiers := range identifiersMap {
+		// Check if there are targeted inventory names and include the current inventory name
+		if _, ok := r.inventoryNameSet[invName]; !ok && len(r.inventoryNameSet) != 0 {
+			continue
+		}
+		// Filter objects
+		for _, obj := range identifiers {
+			// check if the object is under one of the targeted namespaces
+			if _, ok := r.namespaceSet[obj.Namespace]; ok || len(r.namespaceSet) == 0 {
+				// add to the map for future reference
+				printData.InvNameMap[obj] = invName
+				// append to identifiers
+				printData.Identifiers = append(printData.Identifiers, obj)
+			}
+		}
+	}
+	return &printData, nil
+}
+
+// runE implements the logic of the command and will delegate to the
+// poller to compute status for each of the resources. One of the printer
+// implementations takes care of printing the output.
+func (r *Runner) runE(cmd *cobra.Command, args []string) error {
+	var printData *printer.PrintData
+	var err error
+	switch r.invType {
+	case Local:
+		if len(args) != 0 {
+			fmt.Printf("%c[%dm", printcommon.ESC, printcommon.YELLOW)
```

Comment:
I think you can simplify this with:
https://github.com/kubernetes-sigs/cli-utils/blob/eb8ebd73633e0ea5a2cd1406005f828118cef289/pkg/print/common/color.go#L25